aws_list_all/listing.py

```diff
@@ -380,3 +275,59 @@ def resources(self):  # pylint:disable=too-many-branches
             response['truncated'] = [True]
 
         return response
+
+
+class ResultListing(object):
+    """Represents a listing result summary acquired from the function acquire_listing in query.py"""
+    def __init__(self, input, result_type, details, id_list=None, diff=''):
+        self.input = input
+        self.result_type = result_type
+        self.details = details
+        self.id_list = [] if id_list is None else id_list
+        self.diff = diff
+
+    @classmethod
+    def diffInListing(cls, base, diff):
+        return cls(base.input, base.result_type, base.details, base.id_list, diff)
+
+    def __eq__(self, other):
```

Comment:
Here you can just define __eq__ and __lt__ and then use https://docs.python.org/3/library/functools.html#functools.total_ordering to generate the rest